packages/vast-xml2js/src/helpers/xmlToJson.js

```diff
@@ -0,0 +1,66 @@
+const TYPE = {
+  CDATA: 'cdata',
+  DOCUMENT: 'document',
+  ELEMENT: 'element',
+  TEXT: 'text'
+};
+
+const nodeType = (node) => {
```

Comment:
Why not an object mapping?
if you can use @A0dh suggestion:
```javascript
const nodeTypeMap = {
  [node.ELEMENT_NODE]: TYPE.ELEMENT,
};

const type = nodeTypeMap[node.nodeType];
if (!type) throw new Error('Unsupported element type');
return type;
```

Reply:
I used the @A0dh as a suggestion, but if you can't use those constants you can use this
```javascript

const supportedNodeTypes = {
   [1]:  TYPE.ELEMENT,
   [3]: TYPE.TEXT,
   [4]: TYPE.TEXT,
   [9]: TYPE. DOCUMENT
};

const type = supportedNodeTypes[node.nodeType];
if (!type) throw new Error('Unsupported element type');
return type;
```

Reply:
Same question could be, are the numbers you are using: `1, 2, ...` are implemented on every browser? etc.

Reply:
You might be right being careful about this, as DOM4 spec has adjustment with regards to 2, 4, 5, 6, 12 node types constants. 

Ref: https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

Reply:
saw them. thanks